drivers/adc/adc_demo/iio_adc_demo.h

```diff
@@ -42,10 +42,86 @@
 
 #include <stdlib.h>
 #include "adc_demo.h"
+#include "iio_types.h"
 
-struct iio_device adc_demo_iio_descriptor = {
-	.debug_reg_read = (int32_t (*)()) adc_demo_reg_read,
-	.debug_reg_write = (int32_t (*)()) adc_demo_reg_write
+#define DEFAULT_CHANNEL_NO 2
+#define ADC_CHANNEL_NO changeable_channel_number
```

Comment:
I would move this to adc_demo.h as:
#infdef MODIFIED_CHANNEL_VALUE
#define ADC_CHANNEL_NO 2
#else
#define ADC_CHANNEL_NO 16
#endif

Reply:
I wrote them this way so that it can be used with values other than 2 and 16

Reply:
then I think you can do something like this:

**#ifndef ADC_CHANNEL_NO**
#infdef MODIFIED_CHANNEL_VALUE
#define ADC_CHANNEL_NO 2
#else
#define ADC_CHANNEL_NO 16
#endif
**#endif**

This allows you to define custom `ADC_CHANNEL_NO` macro inside your main app directly.

And you can avoid passing static variables to macros, which I don't think is a good practice.